src/main/java/gyro/google/compute/AbstractInstanceGroupManagerResource.java

```diff
@@ -0,0 +1,446 @@
+/*
+ * Copyright 2020, Perfect Sense, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package gyro.google.compute;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import com.google.api.services.compute.model.InstanceGroupManager;
+import com.google.api.services.compute.model.InstanceGroupManagerAutoHealingPolicy;
+import com.google.api.services.compute.model.InstanceGroupManagerVersion;
+import com.google.api.services.compute.model.NamedPort;
+import gyro.core.resource.Output;
+import gyro.core.resource.Updatable;
+import gyro.core.validation.ConflictsWith;
+import gyro.core.validation.Min;
+import gyro.core.validation.Regex;
+import gyro.core.validation.Required;
+import gyro.core.validation.ValidationError;
+import gyro.google.Copyable;
+
+public abstract class AbstractInstanceGroupManagerResource extends ComputeResource
+    implements Copyable<InstanceGroupManager> {
+
+    private String baseInstanceName;
+
+    private String name;
+
+    private Integer targetSize;
+
+    private List<ComputeInstanceGroupManagerAutoHealingPolicy> autoHealingPolicy;
+
+    private String description;
+
+    private String fingerprint;
+
+    private InstanceTemplateResource instanceTemplate;
+
+    private List<ComputeNamedPort> namedPort;
+
+    // TODO: target pool resource
+    // https://github.com/perfectsense/gyro-google-provider/issues/79
+    //    private List<AbstractTargetPoolResource> targetPools;
```

Comment:
Should be Removed and be done part of a separate pr when target pool is available

Reply:
Yeah, it should be handled in a separate PR, but better to leave the comment as it will be helpful when implementing it.